custom_components/sygnal/climate.py

```diff
@@ -67,11 +76,14 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
     # Poll at setup time to get unique id and zone info
     await api.async_update()
 
-    entities = []
+    async_add_entities([SygnalClimate(name, api)])
+
+    # TODO: possible race here if switch does not (yet) exist
+    switch_platform = get_platform(hass, Platform.SWITCH)
```

Comment:
I'm a but rusty with this but the code here looks like it reaches into the switch platform and adds the zone entities to that. Is that what we should be doing here?